tasks/src/tasks/fetch.rs

```diff
@@ -0,0 +1,116 @@
+use std::{collections::HashMap, num::NonZeroU32};
+
+use imap_types::{
+    command::CommandBody,
+    core::Vec1,
+    fetch::{MacroOrMessageDataItemNames, MessageDataItem},
+    response::{Data, StatusBody, StatusKind},
+    sequence::SequenceSet,
+};
+use tracing::warn;
+
+use super::TaskError;
+use crate::Task;
+
+/// Fetch message data items matching the given sequence set and the
+/// given item names (or macro).
+#[derive(Clone, Debug)]
+pub struct FetchTask {
+    sequence_set: SequenceSet,
+    macro_or_item_names: MacroOrMessageDataItemNames<'static>,
+    uid: bool,
+    output: HashMap<NonZeroU32, Vec1<MessageDataItem<'static>>>,
+}
+
+impl Task for FetchTask {
+    type Output = Result<HashMap<NonZeroU32, Vec1<MessageDataItem<'static>>>, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        CommandBody::Fetch {
+            sequence_set: self.sequence_set.clone(),
+            macro_or_item_names: self.macro_or_item_names.clone(),
+            uid: self.uid,
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        if let Data::Fetch { items, seq } = data {
+            if let Some(items) = self.output.insert(seq, items) {
+                warn!(?items, "received duplicate items for message {seq}");
```

Comment:
Should we make `seq` also a structured value?

Reply:
Sth like this?

```diff
- warn!(?items, "received duplicate items for message {seq}");
+ warn!(?items, seq, "received duplicate items");
```

I hesitatetd, somehow I found the non structured version more readable, but I don't have solid argument for. Up to you, what do you think is better?

Reply:
I see why you hesitated. Still, I think I would prefer the structured version.